code/datums/status_effects/drug_effects.dm

```diff
@@ -58,3 +58,39 @@
 	name = "Seizure"
 	desc = "FJOIWEHUWQEFGYUWDGHUIWHUIDWEHUIFDUWGYSXQHUIODSDBNJKVBNKDML <--- this is you right now"
 	icon_state = "paralysis"
+
+/datum/status_effect/stoned
+	id = "stoned"
+	duration = 10 SECONDS
+	alert_type = /atom/movable/screen/alert/status_effect/stoned
+	status_type = STATUS_EFFECT_REFRESH
+	var/original_eye_color
+
+/datum/status_effect/stoned/on_apply()
+	if(ishuman(owner))
+		var/mob/living/carbon/human/human_owner = owner
+		original_eye_color = human_owner.eye_color
+		human_owner.add_movespeed_modifier(/datum/movespeed_modifier/reagent/cannabis) //slows you down
+		human_owner.eye_color = BLOODCULT_EYE //makes cult eyes less obvious
+		human_owner.update_body() //updates eye color
+		ADD_TRAIT(human_owner, TRAIT_BLOODSHOT_EYES, type) //dilates blood vessels in eyes
+		ADD_TRAIT(human_owner, TRAIT_CLUMSY, type) //impairs motor coordination
+		SEND_SIGNAL(human_owner, COMSIG_ADD_MOOD_EVENT, "stoned", /datum/mood_event/stoned) //improves mood
+		human_owner.sound_environment_override = SOUND_ENVIRONMENT_DRUGGED //not realistic but very immersive
+	return TRUE
```

Comment:
```suggestion
	if(!ishuman(owner))
		return FALSE
	var/mob/living/carbon/human/human_owner = owner
	original_eye_color = human_owner.eye_color
	human_owner.add_movespeed_modifier(/datum/movespeed_modifier/reagent/cannabis) //slows you down
	human_owner.eye_color = BLOODCULT_EYE //makes cult eyes less obvious
	human_owner.update_body() //updates eye color
	ADD_TRAIT(human_owner, TRAIT_BLOODSHOT_EYES, type) //dilates blood vessels in eyes
	ADD_TRAIT(human_owner, TRAIT_CLUMSY, type) //impairs motor coordination
	SEND_SIGNAL(human_owner, COMSIG_ADD_MOOD_EVENT, "stoned", /datum/mood_event/stoned) //improves mood
	human_owner.sound_environment_override = SOUND_ENVIRONMENT_DRUGGED //not realistic but very immersive
	return TRUE
```
* Early return and return false if the owner is not human. We don't want to keep the trait then.